Jellyfin.Api/Controllers/DynamicHlsController.cs

```diff
@@ -1503,8 +1503,17 @@ private async Task<ActionResult> GetDynamicSegment(StreamingRequestDto streaming
                         // If the playlist doesn't already exist, startup ffmpeg
                         try
                         {
-                            await _transcodingJobHelper.KillTranscodingJobs(streamingRequest.DeviceId, streamingRequest.PlaySessionId, p => false)
+                            if (_encodingOptions.EnableThrottling && _encodingOptions.EnableSegmentDeletion)
+                            {
+                                // Delete old HLS files when segment deletion is active since ffmpeg doesn't clean them up by itself
+                                await _transcodingJobHelper.KillTranscodingJobs(streamingRequest.DeviceId, streamingRequest.PlaySessionId, p => true)
                                 .ConfigureAwait(false);
+                            }
+                            else
+                            {
+                                await _transcodingJobHelper.KillTranscodingJobs(streamingRequest.DeviceId, streamingRequest.PlaySessionId, p => false)
+                                    .ConfigureAwait(false);
+                            }
```

Comment:
Simplification. Wouldn't it be safe to delete regardless though?
```suggestion
                            // Delete old HLS files when segment deletion is active since ffmpeg doesn't clean them up by itself
                            var deleteFiles = _encodingOptions.EnableThrottling && _encodingOptions.EnableSegmentDeletion;
                            await _transcodingJobHelper.KillTranscodingJobs(streamingRequest.DeviceId, streamingRequest.PlaySessionId, p => deleteFiles)
                                .ConfigureAwait(false);
```

Reply:
> Wouldn't it be safe to delete regardless though?

Not entirely sure, I didn't want to interfere too much with the regular transcoding method since I don't know if the old segments are re-used when scrubbing or not.